scripts/devenv.py

```diff
@@ -1,272 +1,94 @@
-    sub = ap.add_subparsers(dest="command", metavar="COMMAND")
-    sub.required = True
+    if not all(map(shutil.which, EXECUTABLES)):
+        sys.exit("executables " + punctlist(EXECUTABLES) + " should exist in the system path")
 
-    create_ap = sub.add_parser(
-        "create", help="create a new development environment", formatter_class=argparse.ArgumentDefaultsHelpFormatter
-    )
-    create_ap.set_defaults(func=command_create)
-    add_global_args(create_ap)
-    create_ap.add_argument(
-        "-p",
-        "--package",
-        help="conda packages to install",
-        metavar="PKG",
-        nargs="*",
-        default=["ilastik-dependencies-no-solvers"],
-        action=ReplaceAction,
-    )
-    create_ap.add_argument(
-        "-c",
-        "--channel",
-        help="conda channels to use",
-        metavar="CHAN",
-        nargs="*",
-        default=["ilastik-forge", "conda-forge"],
-        action=ReplaceAction,
-    )
-    create_ap.add_argument(
-        "-l",
-        "--location",
-        help="local ilastik-meta directory",
-        metavar="DIR",
-        nargs="?",
-        const=None,
-        default=str(pathlib.Path(".").resolve()),
-    )
+    if not all(map(is_git_repo, REPOSITORIES)):
+        sys.exit("directories " + punctlist(REPOSITORIES) + " should be valid git repositories")
 
-    remove_ap = sub.add_parser(
-        "remove",
-        help="remove an existing development environment",
-        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
-    )
-    remove_ap.set_defaults(func=command_remove)
-    add_global_args(remove_ap)
+    logging.info("command %r started", args.cmd)
+    time_start = time.perf_counter()
 
-    args = ap.parse_args()
+    try:
+        args.func(args)
+    except subprocess.CalledProcessError:
+        logging.error("command %r failed", args.cmd)
+    except KeyboardInterrupt:
+        logging.error("command %r interrupted", args.cmd)
+    else:
+        logging.info("command %r completed", args.cmd)
 
-    if shutil.which("conda") is None:
-        logging.error("conda is not installed")
-        return
+    time_end = time.perf_counter()
+    logging.info("elapsed time: %.3f seconds", time_end - time_start)
 
-    if run_returncode("conda", "build", "--version") != 0:
-        logging.error("conda-build is not installed; run 'conda install -n base -c conda-forge conda-build' to install")
-        return
 
-    env = CondaEnv(args.name)
-    execution_time, _res = timed(args.func, args, env)
-    logging.info(f"finished {args.command} ({execution_time:.3f} seconds)")
+def action_create(args):
+    run(["conda", "install", "--yes", "--name", "base", "--channel", "conda-forge", "conda-build"])
 
+    chans = ["--channel", "ilastik-forge", "--channel", "conda-forge"]
+    pkgs = ["ilastik-dependencies-no-solvers", "pre-commit"]
+    run(["conda", "create", "--yes", "--name", args.name] + chans + pkgs)
```

Comment:
should have `--override-channels` and `--strict-channel-priority` (just in case someone has a `.condarc` stating something else.